File: cc/array/PositionsOfLargeGroups.cc
#include <vector>
#include <string>

using namespace std;

/**
 * LeetCode 830. Positions of Large Groups
 * https://leetcode.com/problems/positions-of-large-groups/
 */
class PositionsOfLargeGroups {
public:
    vector<vector<int>> largeGroupPositions(string s) {
        vector<vector<int>> vec;
        int startIdx = 0;
        char c = 0;
        int n = s.size();
        for (int i = 0; i < n; /* ... */) {
            startIdx = i;
            c = s[startIdx];
            while (i < n && s[i] == c) {
                i++;
            }
            if (i - startIdx >= 3) {
                vector<int> v;
                v.push_back(startIdx);
                v.push_back(i - 1);
                vec.push_back(v);
            }
        }
        return vec;
    }
};

int main(int argc, char const *argv[]) {
    // ...
    return 0;
}

/* EOF */

File: java/src/io/zihengcat/leetcode/array/PositionsOfLargeGroups.java
package io.zihengcat.leetcode.array;

import java.util.List;
import java.util.ArrayList;

/**
 * LeetCode 830. Positions of Large Groups
 * https://leetcode.com/problems/positions-of-large-groups/
 */
public class PositionsOfLargeGroups {
    public static void main(String[] args) {
        // ...
    }
    public List<List<Integer>> largeGroupPositions(String s) {
        if (s == null || s.length() == 0) {
            return null;
        }
        List<List<Integer>> aList = new ArrayList<List<Integer>>();
        int sIdx = 0;
        char c = 0;
        int n = s.length();
        for (int i = 0; i < n; /* ... */) {
            sIdx = i;
            c = s.charAt(sIdx);
            while (i < n && s.charAt(i) == c) {
                i++;
            }
            if (i - sIdx >= 3) {
                List<Integer> arr = new ArrayList<Integer>();
                arr.add(sIdx);
                arr.add(i - 1);
                aList.add(arr);
            }
        }
        return aList;
    }
}

/* EOF */

